From: Gregory M. Kurtzer <gmkurtzer@lbl.gov>
Subject: Fix busybox path

--- a/examples/busybox.def
+++ b/examples/busybox.def
@@ -5,7 +5,7 @@
 # required approvals from the U.S. Dept. of Energy).  All rights reserved.
 
 BootStrap: busybox
-MirrorURL: https://www.busybox.net/downloads/binaries/busybox-x86_64
+MirrorURL: https://www.busybox.net/downloads/binaries/1.26.2-defconfig-multiarch/busybox-x86_64
 
 
 %post
